### ISSUE TITLE
Casting strings to JSONB silently ignores syntax errors

### ISSUE THREAD

**Issue by petrosagg** [MaterializeInc/materialize#5829]
### What was the issue?

Behavior deviates from postgres behavior.

Postgres:

```bash
postgres=# SELECT 'fdsf'::jsonb;
ERROR:  invalid input syntax for type json
LINE 1: SELECT 'fdsf'::jsonb;
               ^
DETAIL:  Token "fdsf" is invalid.
CONTEXT:  JSON data, line 1: fdsf
```

Materialize:

```bash
materialize=> SELECT 'fdsf'::jsonb;
 ?column?
----------

(1 row)
```

**Comment by rjnn** [comment]
Probably still logged, right? Seems like a runtime error reporting issue (cc @benesch who would know)

**Comment by benesch** [comment]
Petros already has a fix out in #5830!
